lucene/core/src/java/org/apache/lucene/store/FileSwitchDirectory.java

```diff
@@ -141,7 +147,14 @@ public static String getExtension(String name) {
     if (i == -1) {
       return "";
     }
-    return name.substring(i+1, name.length());
+    String ext = name.substring(i + 1);
+    if (ext.equals("tmp")) {
+      Matcher matcher = EXT_PATTERN.matcher(name);
+      if (matcher.find()) {
+        return matcher.group(1);
+      }
```

Comment:
Hmm should we throw exception in `else` clause here?  Force callers to use consistent `tmp` file naming?

Reply:
that's difficult since it's ok to create a tmp file with a prefix of "foo" and a suffix of "bar" that will result in no match. I think we can only do the best effort on the rename end. I pushed a new commit to make the heuristic better. I thin that's all we can do.

Reply:
OK